### PR TITLE
Add DatabaseService

### PR DESCRIPTION
**Pull request by MacMalainey**
Add initial database service with capability to query specific documents.

## Description of Changes
Add initial database service
Add ability to query specific documents
Add ability to select specific collections
Add ability to parse database objects to specific model objects

## Checklist

- [ ] Automated tests
- [x] Screenshots included (if necessary)
- [x] Code is well commented

### CODE REVIEW

**Review by AtrazineC — COMMENTED** [review]
Can you provide a few examples of what this will be used for?

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/services/database/DatabaseService.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.cmput301f21t09.budgetprojectname.services.database;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+
```
Comment (j-vent) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/services/database/DatabaseService.java:4]: Add documentation and specify this is a singleton design pattern

**Review by j-vent — DISMISSED** [review]
I'm still reading up on listeners so I'm just gonna trust that the implementation is correct.

**Comment by j-vent** [comment]
+1 Ben's comment. I just want to approve so we're not blocked but also in the meeting it would be good to go over use cases specifically how it will interact w stores

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/services/database/DatabaseService.java (hunk 1 above)
Comment (MacMalainey) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/services/database/DatabaseService.java:4]: This should never be a singleton.  Firebase Firestore is connected to the application context and therefore any instance of it should never be stored in global scope.

**Comment by MacMalainey** [comment]
This will provide an abstraction layer for interacting with the database backend.  In production quality code we should never have the UI components interface directly with the backend API.

So it will handle any and all writes/reads to/from the database.  I just have not implemented writes yet or more specific queries.

**Review by Docker-J — APPROVED** [review]

**Review by cheetiantey — DISMISSED** [review]

**Comment by MacMalainey** [comment]
Notes on the new update:
It is not fully commented
This is just an example of a "fully" defined MVC view

**Pull request closed without merging**